src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20240103_143700.java

```diff
@@ -0,0 +1,415 @@
+            Optional<BitbucketLocalVCMigrationService> bitbucketLocalVCMigrationService) {
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.environment = environment;
+        this.solutionProgrammingExerciseParticipationRepository = solutionProgrammingExerciseParticipationRepository;
+        this.templateProgrammingExerciseParticipationRepository = templateProgrammingExerciseParticipationRepository;
+        this.programmingExerciseStudentParticipationRepository = programmingExerciseStudentParticipationRepository;
+        this.localVCService = localVCService;
+        this.ciMigrationService = ciMigrationService;
+        this.gitService = gitService;
+        this.auxiliaryRepositoryRepository = auxiliaryRepositoryRepository;
+        this.bitbucketService = bitbucketService;
+        this.bitbucketLocalVCMigrationService = bitbucketLocalVCMigrationService;
+    }
+
+    @Override
+    public void execute() {
+        List<String> activeProfiles = List.of(environment.getActiveProfiles());
+        if (!new HashSet<>(activeProfiles).containsAll(MIGRATABLE_PROFILES)) {
+            log.info("Migration will be skipped and marked run because the system does not support a tech-stack that requires this migration: {}",
+                    List.of(environment.getActiveProfiles()));
+            return;
+        }
+
+        if (bitbucketLocalVCMigrationService.isEmpty()) {
+            log.error("Migration will be skipped and marked run because the BitbucketLocalVCMigrationService is not available.");
+            return;
+        }
+
+        if (bitbucketLocalVCMigrationService.get().getLocalVCBasePath() == null) {
+            log.error("Migration will be skipped and marked run because the local VC base path is not configured.");
+            return;
+        }
+
+        if (bitbucketLocalVCMigrationService.get().getLocalVCBaseUrl() == null || bitbucketLocalVCMigrationService.get().getLocalVCBaseUrl().toString().equals("http://0.0.0.0")) {
+            log.error("Migration will be skipped and marked run because the local VC base URL is not configured.");
+            return;
+        }
+
+        if (bitbucketLocalVCMigrationService.get().getDefaultBranch() == null) {
+            log.error("Migration will be skipped and marked run because the default branch is not configured.");
+            return;
+        }
+
+        var programmingExerciseCount = programmingExerciseRepository.count();
+        var studentCount = ciMigrationService.orElseThrow().getPageableStudentParticipations(programmingExerciseStudentParticipationRepository, Pageable.unpaged())
+                .getTotalElements();
+
+        if (programmingExerciseCount == 0) {
+            // no exercises to change, migration complete
+            return;
+        }
+
+        log.info("Will migrate {} programming exercises and {} student repositories now. This might take a while", programmingExerciseCount, studentCount);
```

Comment:
please add a rough time estimate based on the count. I think we can assume that a migration of one repository will take around 2s. Take the number of threads into account